src/main.ts

```diff
@@ -1,169 +1,293 @@
 import * as core from '@actions/core'
```

Comment:
I'm gonna give it to you straight, this 300 line file stuffed with 50 line functions is kind of scary. It probably looks clearer in an IDE when you can ctrl+click around, but I feel like this'd just benefit from at least breaking up the functions into different files. I'd even make a case for trying to find a more object-oriented approach to doing this, but that's a way bigger refactor, so I'm perfectly fine to settle for just breaking this up a little bit.

Reply:
How's that?